lib/Locale/Country/Extra.pm

```diff
@@ -46,21 +47,19 @@ sub idd_from_code {
 }
 
 sub code_from_phone {
-    my ( $self, $number ) = @_;
+    my ($self, $number) = @_;
 
     $number =~ s/\D//g;    # Remove non-digits
     $number =~ s/^00//;    # Remove the leading '00'.
 
-    if ( $number !~
-/^(111111|222222|333333|444444|555555|666666|777777|888888|999999|000000)/
-      )
-    {
-        my %code_for_idd = reverse %{ $self->_idd_codes };
-        foreach my $iddcode ( sort { $b <=> $a } keys %code_for_idd ) {
-            if ( $number =~ /^$iddcode/ ) {
-                return lc $code_for_idd{$iddcode};
-            }
+    if ($number !~ /^(111111|222222|333333|444444|555555|666666|777777|888888|999999|000000)/) {
+        my @country_list;
+        my %codes = %{$self->_idd_codes};
+        for my $key (keys %codes) {
+            push(@country_list, $key) if $number =~ /^$codes{$key}/;
         }
+        @country_list = sort @country_list;
+        return \@country_list;
```

Comment:
this is changing the API? It's a public module, we can't do that without a major version increase and a big warning in the change log.

I would recommend making a separate method and having this preserve the original single-value behaviour.

Also, I think that entire expression is basically just trying to do `return [ sort grep { $number =~ /\Q$codes{$_}/ } keys %codes ]` ?

Reply:
@binarytom we are not using this call on the API, the change on this method is required because of the duplicated codes like +44, instead of return just `gb` now, we need to return `im` too.